### PR TITLE
feat(argo-applicationset): update appVersion to 0.4.1

### PR DESCRIPTION
**Pull request by pburgisser**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [x] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

File: charts/argocd-applicationset/Chart.yaml [hunk 1]
```diff
@@ -14,4 +14,5 @@ maintainers:
   - name: maruina
 annotations:
   artifacthub.io/changes: |
+    - "[Changed]: updated application version for v0.4.1 and its CRD for support of skipCrds in applications"
     - "[Added]: extraContainers support in Deployment resource"
```
Comment (mbevc1) [inline charts/argocd-applicationset/Chart.yaml:18]: Please remove old changelog as it get appended

**Review by mkilchhofer — APPROVED** [review]
Its a bit in conflict with https://github.com/argoproj/argo-helm/issues/1219. Maybe after we merge this change here, we deprecated the charts directly and only maintain it in argo-cd chart directly

**Comment by pburgisser** [comment]
Indeed, I realized that when I tested it in production with an old version of the Argo-cd helm chart :)

File: charts/argocd-applicationset/templates/deployment.yaml [hunk 2]
```diff
@@ -33,6 +33,7 @@ spec:
           securityContext:
             {{- toYaml .Values.securityContext | nindent 12 }}
           command:
+            - entrypoint.sh
```
Comment (mkilchhofer) [inline charts/argocd-applicationset/templates/deployment.yaml:36]: Synced this change also from the upstream manifests

**Pull request merged**